commands/explorepotatoes.js

```diff
@@ -104,56 +104,42 @@ const info = {
 				.addChannelTypes([0, 4, 5, 6, 10, 11]),
 		),
 
-	async interaction(interaction) {
-		if (interaction.guild?.id !== process.env.GUILD_ID) return;
-		const deferPromise = interaction.deferReply();
-		const board = await interaction.guild?.channels.fetch(BOARD_CHANNEL);
+	async interaction(_interaction) {
+		// Never name a param in a function the name of the same function, you'll run into tricky bugs
+		if (_interaction.guild?.id !== process.env.GUILD_ID) return;
+		const deferPromise = _interaction.deferReply();
+		const board = await _interaction.guild?.channels.fetch(BOARD_CHANNEL);
 		if (!board?.isText())
 			throw new Error(
 				"No board channel found. Make sure BOARD_CHANNEL is set in the .env file.",
 			);
 
-		const minReactions = interaction.options.getInteger("minimum-reactions") || 0;
-		const user = interaction.options.getUser("user")?.id;
-		const channelId = interaction.options.getChannel("channel")?.id;
-		const channelWanted = channelId && (await interaction.guild?.channels.fetch(channelId));
+		const minReactions = _interaction.options.getInteger("minimum-reactions") || 0;
+		const user = _interaction.options.getUser("user")?.id;
+		const channelId = _interaction.options.getChannel("channel")?.id;
+		const channelWanted = channelId && (await _interaction.guild?.channels.fetch(channelId));
 		const [, fetchedMessages] = await Promise.all([
 			deferPromise,
-			getAllMessages(board).then((messages) =>
-				asyncFilter(messages, async (message) => {
-					if (!message.content || !message.embeds[0] || !message.author.bot) return false;
-					if ((message.content.match(/\d+/)?.[0] || 0) < minReactions) return false;
-					if (user && message.mentions.users.first()?.id !== user) return false;
-
-					if (channelWanted) {
-						const matchResult = message.content.match(/<#(\d+)>/g);
-						const channelFound = message.mentions.channels.first() || matchResult?.[1];
-
-						if (
-							!(
-								channelFound &&
-								(await textChannelMatchesChannel(channelWanted, channelFound))
-							)
-						) {
-							return false;
-						}
-					}
-					return true;
-				}),
-			),
+			// Organize code a bit
+			_getMessages(board)
 		]);
-
+		// Shuffle
+		fetchedMessages = fetchedMessages.sort(() => 0.5 - Math.random())
```

Comment:
I shuffled the list initially instead of getting a random index every time. Before this PR the next button just showed a random item in `fetchedMessages`. Now it has the same effect, but the back button shows the last random message